django/db/models/fields/related.py

```diff
@@ -613,6 +613,21 @@ def resolve_related_fields(self):
             )
             to_field = (self.remote_field.model._meta.pk if to_field_name is None
                         else self.remote_field.model._meta.get_field(to_field_name))
+            if (
+                to_field_name is not None and to_field_name != 'id' and
+                not self.remote_field.model._meta.proxy and
```

Comment:
This condition added for `to_field` which is pointing to the proxy model. for example:
https://github.com/django/django/blob/1a9459b88e4b07b3f4dd80ed5987e13111afe1c2/tests/proxy_models/models.py#L168

Reply:
Use `self.remoted_field.model._meta.concrete_model` below instead